pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -284,41 +284,82 @@ func add(mgr, adminMgr manager.Manager, r reconcile.Reconciler) error {
 				Kind:       "Namespace",
 				APIVersion: "v1",
 			},
-		}),
-		namespaceMapFn,
-		namespacePred)
+		},
+			namespaceMapFn,
+			namespacePred))
 	if err != nil {
 		return err
 	}
 
+	// configMapAllCredRequestsMapFn simply looks up all CredentialsRequests and requests they be reconciled.
+	configMapAllCredRequestsMapFn := handler.TypedEnqueueRequestsFromMapFunc(func(ctx context.Context, a *corev1.ConfigMap) []reconcile.Request {
+		log.Info("requeueing all CredentialsRequests")
+		crs := &minterv1.CredentialsRequestList{}
+		err := mgr.GetClient().List(ctx, crs)
+		var requests []reconcile.Request
+		if err != nil {
+			log.WithError(err).Error("error listing all cred requests for requeue")
+			return requests
+		}
+		for _, cr := range crs.Items {
+			requests = append(requests, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      cr.Name,
+					Namespace: cr.Namespace,
+				},
+			})
+		}
+		return requests
+	})
+
 	// Monitor the cloud-credential-operator-config configmap to check if operator is re-enabled.
 	// Check if attempts were made to delete the credentials requests when the operator was disabled.
 	// We have to have to reconcile all those credential requests.
 	// Check if operator is re-enabled in cloud-credential-operator-config configmap
-	configMapPredicate := predicate.Funcs{
-		UpdateFunc: func(e event.UpdateEvent) bool {
+	configMapPredicate := predicate.TypedFuncs[*corev1.ConfigMap]{
+		UpdateFunc: func(e event.TypedUpdateEvent[*corev1.ConfigMap]) bool {
 			return isCloudCredOperatorConfigMap(e.ObjectNew)
 		},
-		CreateFunc: func(e event.CreateEvent) bool {
+		CreateFunc: func(e event.TypedCreateEvent[*corev1.ConfigMap]) bool {
 			return isCloudCredOperatorConfigMap(e.Object)
 		},
-		DeleteFunc: func(e event.DeleteEvent) bool {
+		DeleteFunc: func(e event.TypedDeleteEvent[*corev1.ConfigMap]) bool {
 			return isCloudCredOperatorConfigMap(e.Object)
 		},
 	}
 	err = c.Watch(
-		source.Kind(operatorCache, &corev1.ConfigMap{}),
-		allCredRequestsMapFn,
-		configMapPredicate)
+		source.Kind(operatorCache, &corev1.ConfigMap{},
+			configMapAllCredRequestsMapFn,
+			configMapPredicate))
 	if err != nil {
 		return err
 	}
 
+	// configMapAllCredRequestsMapFn simply looks up all CredentialsRequests and requests they be reconciled.
+	credentialMapAllCredRequestsMapFn := handler.TypedEnqueueRequestsFromMapFunc(func(ctx context.Context, a *operatorv1.CloudCredential) []reconcile.Request {
```

Comment:
I think this function could be made generic and combined with the similarly named function on line 295.  Neither function actually references the `a` parameter anyways

Reply:
They don't reference the parameter, but the parameter is of different type.

Reply:
I wanted to break this out into its own function, but they use `mgr` inline, which can't really be passed any other way.